### PR TITLE
[Dynamo] add EventVariable reconstruct

### PR DESCRIPTION
**Pull request by yf225**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #133236

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames

### CODE REVIEW

File: test/dynamo/test_ctx_manager.py [hunk 1]
```diff
@@ -361,6 +360,30 @@ def fn(x, s0, s1):
...
+    def test_cuda_event_across_graph_break(self):
+        def fn(x):
+            e = torch.cuda.Event()
+            e.record()
+            x = torch.mul(x, 5)
+            x = torch.add(x, 2)
+
+            print("foo")
+
+            e.wait()
```
Comment (yifuwang) [inline test/dynamo/test_ctx_manager.py:373]: Is this wait going to be on the same event?

Comment (yf225) [reply test/dynamo/test_ctx_manager.py]: reduced scope of PR - actually only want to add the EventVariable reconstruct for now

**Review by yifuwang — APPROVED** [review]

**Comment by yf225** [comment]
I am not sure why, but it seems that this PR is not auto-closed after being landed as https://github.com/pytorch/pytorch/commit/1206958d89ffa510243e6f534d535942b791535f. So manually closing it here.

**Pull request closed without merging**